dynamiqs/mesolve/rouchon.py

```diff
@@ -91,13 +91,27 @@ def backward_augmented(self, t: float, rho: Tensor, phi: Tensor):
 
 
 class MERouchon1_5(MERouchon):
-    # TODO implement caching
-    def forward(self, t: float, rho: Tensor) -> Tensor:
-        r"""Compute $\rho(t+dt)$ using a Rouchon method of order 1.5.
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-        Note:
-            No need for trace renormalization since the scheme is trace-preserving
-            by construction.
+        # compute the inverse square root renormalization matrix
+        # (for time-dependent Hamiltonians, exclude the Hamiltonian from normalization)
+        if isinstance(self.H, ConstantTDTensor):
+            H_nh = self.H_nh(0.0)  # (b_H, 1, n, n)
+        elif isinstance(self.H, CallableTDTensor):
+            H_nh = -0.5j * self.sum_no_jump  # (b_H, 1, n, n)
```

Comment:
Where did you see this? Are you sure it's ok to exclude the Hamiltonian from the normalisation?

Reply:
Yes essentially it's better than nothing. Recomputing the normalization at every time step would be extremely costly, and it's often the case that numerical divergence is caused by the Ldag @ L terms that explode as the Fock truncature gets larger.